docs/quality-assurance-test-cases.md

```diff
@@ -156,6 +156,22 @@ This document contains some examples of test cases for each feature implemented
 | User2 (received the request) cancels a mentorship relation that it is currently involved with User1 (the relation is in an ACCEPTED state)  | Success |
 | User1 cancel a mentorship relation which the User1 is not involved with | Fail |
 
+### Update task
+
+**Service:** PUT /mentorship_relation/{request_id}/task/{task_id}/complete
+
+| Test Case       | Outcome |
```

Comment:
When describing a test case, I think you should only have one failing variable instead of more than one. So for example, if we have 3 variables: a, b and c, in one test case you should describe only one variable failing. This is because what determines the success of the API is if the 3 variables are valid. so it can be redundant saying "a and b are wrong then the API fails", because it takes one of the them to fail for the whole test to fail.

So having your proposal of test cases as examples, you can remove those in which you have for example wrong `token` and wrong `request_id`, because it takes token to be wrong for it to completely fail. Then if you have the 2 wrong, you will fail as well. I will provide specific change request, hopefully this will clarify what I am trying to exaplain.

Reply:
also you could be more specific in what wrong means. So I can see 2 definitions of wrong `request_id`:
- `request_id` is wrong if from a request that neither the mentor or mentee belong to, if the user logged is neither mentor or mentee of the relation.
- `request_id` is wrong if the request is not in the ACCEPTED state, even if the mentor and mentee belong to the relation
- `request_id` is wrong if its not an integer
- `request_id` is wrong if it does not exist at all

If none of these ^^ as true, then the `request_id` is valid

Reply:
Same goes to `task_id`:
- `task_id` is wrong if the task exists but is already completed?!
- `task_id` is wrong if the task with this id does not exist

Otherwise, the `task_id` is valid (correct)

Reply:
Can you update the table and instead of saying just X variable say, what it means in that case being wrong? I will provide one example

Reply:
Maybe you could add before the table the definition of a valid relation and a valid task